webapp/channels/src/components/message_submit_error.tsx

```diff
@@ -1,68 +1,60 @@
 
 import {isErrorInvalidSlashCommand} from 'utils/post_utils';
 
-interface MessageSubmitErrorProps {
-    error: ServerError;
-    handleSubmit: (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+interface Props {
+    error: ServerError | null;
+    handleSubmit: MouseEventHandler<HTMLAnchorElement>;
     submittedMessage?: string;
 }
 
-class MessageSubmitError extends React.PureComponent<MessageSubmitErrorProps> {
-    public renderSlashCommandError = (): string | ReactFragment => {
-        if (!this.props.submittedMessage) {
-            return this.props.error.message;
-        }
-
-        const command = this.props.submittedMessage.split(' ')[0];
-        return (
-            <React.Fragment>
-                <FormattedMessage
-                    id='message_submit_error.invalidCommand'
-                    defaultMessage="Command with a trigger of ''{command}'' not found. "
-                    values={{
-                        command,
-                    }}
-                />
-                <a
-                    href='#'
-                    onClick={this.props.handleSubmit}
-                >
-                    <FormattedMessage
-                        id='message_submit_error.sendAsMessageLink'
-                        defaultMessage='Click here to send as a message.'
-                    />
-                </a>
-            </React.Fragment>
-        );
-    };
-
-    public render(): JSX.Element | null {
-        const error = this.props.error;
-
-        if (!error) {
-            return null;
-        }
+function MessageSubmitError(props: Props) {
+    if (!props.error) {
+        return null;
+    }
 
-        let errorContent: string | ReactFragment = error.message;
-        if (isErrorInvalidSlashCommand(error)) {
-            errorContent = this.renderSlashCommandError();
-        }
+    if (isErrorInvalidSlashCommand(props.error)) {
+        const slashCommand = props.submittedMessage?.split(' ')[0];
 
         return (
             <div className='has-error'>
                 <label className='control-label'>
-                    {errorContent}
+                    <FormattedMessage
+                        id='message_submit_error.invalidCommand'
+                        defaultMessage="Command with a trigger of ''{command}'' not found. "
+                        values={{
+                            slashCommand,
+                        }}
+                    />
+                    <a
+                        href='#'
+                        onClick={props.handleSubmit}
+                    >
+                        <FormattedMessage
+                            id='message_submit_error.sendAsMessageLink'
+                            defaultMessage='Click here to send as a message.'
+                        />
+                    </a>
                 </label>
             </div>
         );
     }
+
+    if (props.error?.message?.trim()?.length === 0) {
```

Comment:
what happen when `props.error?` is null or undefined

Reply:
error is never null, this component is only rendered when error is not null